tests/integration/__tests__/cache.js

```diff
@@ -0,0 +1,338 @@
+        expect.arrayContaining([
+          expect.stringMatching(/^a\.js .+ms$/),
+          expect.stringMatching(/^b\.js .+ms$/),
+        ])
+      );
+
+      // Update `a.js`
+      await fs.writeFile(path.join(dir, "a.js"), "const a = `a`;");
+
+      const { stdout: secondStdout } = await runPrettier(dir, [
+        "--cache",
+        "--write",
+        ".",
+      ]);
+      expect(stripAnsi(secondStdout).split("\n").filter(Boolean)).toEqual(
+        // the cache of `b.js` is only available.
+        expect.arrayContaining([
+          expect.stringMatching(/^a\.js .+ms$/),
+          expect.stringMatching(/^b\.js .+ms \(cached\)$/),
+        ])
+      );
+    });
+
+    it("re-formats when timestamp has been updated", async () => {
+      const { stdout: firstStdout } = await runPrettier(dir, [
+        "--cache",
+        "--write",
+        ".",
+      ]);
+      expect(stripAnsi(firstStdout).split("\n").filter(Boolean)).toEqual(
+        expect.arrayContaining([
+          expect.stringMatching(/^a\.js .+ms$/),
+          expect.stringMatching(/^b\.js .+ms$/),
+        ])
+      );
+
+      // update timestamp
+      const time = new Date();
+      await fs.utimes(path.join(dir, "a.js"), time, time);
+
+      const { stdout: secondStdout } = await runPrettier(dir, [
+        "--cache",
+        "--write",
+        ".",
+      ]);
+      expect(stripAnsi(secondStdout).split("\n").filter(Boolean)).toEqual(
+        // the cache of `b.js` is only available.
+        expect.arrayContaining([
+          expect.stringMatching(/^a\.js .+ms$/),
+          expect.stringMatching(/^b\.js .+ms \(cached\)$/),
+        ])
+      );
+    });
+
+    it("re-formats when options has been updated.", async () => {
+      const { stdout: firstStdout } = await runPrettier(dir, [
+        "--cache",
+        "--write",
+        ".",
+      ]);
+      expect(stripAnsi(firstStdout).split("\n").filter(Boolean)).toEqual(
+        expect.arrayContaining([
+          expect.stringMatching(/^a\.js .+ms$/),
+          expect.stringMatching(/^b\.js .+ms$/),
+        ])
+      );
+
+      const { stdout: secondStdout } = await runPrettier(dir, [
+        "--cache",
+        "--write",
+        "--trailing-comma",
+        "all",
+        ".",
+      ]);
+      expect(stripAnsi(secondStdout).split("\n").filter(Boolean)).toEqual(
+        expect.arrayContaining([
+          expect.stringMatching(/^a\.js .+ms$/),
+          expect.stringMatching(/^b\.js .+ms$/),
+        ])
+      );
+    });
+
+    it("removes cache file when run Prettier without `--cache` option", async () => {
+      await runPrettier(dir, ["--cache", "--write", "."]);
+      await expect(fs.stat(defaultCacheFile)).resolves.not.toThrowError();
+      await runPrettier(dir, ["--write", "."]);
+      await expect(fs.stat(defaultCacheFile)).rejects.toThrowError();
+    });
+  });
+
+  describe("--cache-strategy content", () => {
```

Comment:
Maybe I missed the test, but what happens if I provide _only_ `--cache-strategy` and not `--cache`?

Reply:
[a4b0ab8](https://github.com/prettier/prettier/pull/12800/commits/a4b0ab82adaeb4cec5589d8dce38731b458fe660)

Reply:
Conflict flags are checked in `index.js`, we can also use `context.argv.__raw` instead of removing the default value.

Reply:
How can we? `context.argv.__raw` has default values.

Reply:
You are right, we can't unless we check `rawArguments`, not important anyway.